tests/test_api.py

```diff
@@ -241,33 +243,83 @@ def test_metadata_base(self):
+        self.assertEqual(metafile_obj.to_dict(), data)
+
+        # Test from_dict and to_dict without length.
+        del data["length"]
+        metafile_obj = MetaFile.from_dict(copy.copy(data))
+        self.assertEqual(metafile_obj.to_dict(), data)
+
+        # Test from_dict and to_dict without length and hashes.
+        del data["hashes"]
+        metafile_obj = MetaFile.from_dict(copy.copy(data))
+        self.assertEqual(metafile_obj.to_dict(), data)
+
+
+    def test_targetfile_class(self):
+        # Test from_dict and to_dict with all attributes.
+        data = {
+            "custom": {
+                "file_permissions": "0644"
+            },
+            "hashes": {
+                "sha256": "65b8c67f51c993d898250f40aa57a317d854900b3a04895464313e48785440da",
+                "sha512": "467430a68afae8e9f9c0771ea5d78bf0b3a0d79a2d3d3b40c69fde4dd42c461448aef76fcef4f5284931a1ffd0ac096d138ba3a0d6ca83fa8d7285a47a296f77"
+            },
+            "length": 31
+        }
+        targetfile_obj = TargetFile.from_dict(copy.copy(data))
+        self.assertEqual(targetfile_obj.to_dict(), data)
+
+        # Test from_dict and to_dict without custom.
+        del data["custom"]
+        targetfile_obj = TargetFile.from_dict(copy.copy(data))
+        self.assertEqual(targetfile_obj.to_dict(), data)
+
+
     def test_metadata_snapshot(self):
         snapshot_path = os.path.join(
                 self.repo_dir, 'metadata', 'snapshot.json')
         snapshot = Metadata.from_file(snapshot_path)
 
         # Create a dict representing what we expect the updated data to be
-        fileinfo = copy.deepcopy(snapshot.signed.meta)
+        fileinfo = copy.deepcopy(snapshot.signed.meta['role1.json'])
         hashes = {'sha256': 'c2986576f5fdfd43944e2b19e775453b96748ec4fe2638a6d2f32f1310967095'}
-        fileinfo['role1.json']['version'] = 2
-        fileinfo['role1.json']['hashes'] = hashes
-        fileinfo['role1.json']['length'] = 123
-
-
-        self.assertNotEqual(snapshot.signed.meta, fileinfo)
-        snapshot.signed.update('role1', 2, 123, hashes)
-        self.assertEqual(snapshot.signed.meta, fileinfo)
+        fileinfo.version = 2
+        fileinfo.hashes = hashes
+        fileinfo.length = 123
+
+        self.assertNotEqual(
+            snapshot.signed.meta['role1.json'].to_dict(), fileinfo.to_dict()
+        )
+        snapshot.signed.update('role1', fileinfo)
+        self.assertEqual(
+            snapshot.signed.meta['role1.json'].to_dict(), fileinfo.to_dict()
+        )
 
         # Update only version. Length and hashes are optional.
-        snapshot.signed.update('role2', 3)
-        fileinfo['role2.json'] = {'version': 3}
-        self.assertEqual(snapshot.signed.meta, fileinfo)
+        fileinfo = MetaFile(3)
+        snapshot.signed.update('role2', fileinfo)
+        self.assertEqual(
+            snapshot.signed.meta['role2.json'].to_dict(), fileinfo.to_dict()
```

Comment:
Just a nit but mentioning in case there's something missing: This test doesn't really test much anything anymore (it used to compare two separate dicts, now it's pretty much comparing an object to itself)

Reply:
Yes, I understand this. The test by itself makes sure that we are actually supporting optional `hashes` and `length`.
Will remove the test in the `fix update argument` commit. Then, we won't need this, because length and hashes will be optional because of the `MetaFile` implementation, no the update function itself.